packages/page-staking/src/Nomination/index.tsx

```diff
@@ -0,0 +1,380 @@
+      if (currentStep === steps[2]) {
+        setCurrentStep(steps[3]);
+      }
+    }
+    if (isNominated) {
+      setStepsState((prevState): string[] => {
+        const newState = [...prevState];
+        newState[2] = 'completed';
+        newState[3] = 'completed';
+        return newState;
+      });
+      // go to last tab
+      if (currentStep === steps[2]) {
+        setCurrentStep(steps[3]);
+      }
+    }
+  }
+
+  function balanceWrapper(text: string): React.ReactNode {
+    return (
+      <strong className='label'>{text}</strong>
+    );
+  }
+
+  function goBack() {
+    const ind = steps.indexOf(currentStep);
+    setCurrentStep(steps[ind - 1]);
+  }
+
+  function goNext() {
+    const ind = steps.indexOf(currentStep);
+    if (ind < 3) {
+      setCurrentStep(steps[ind + 1]);
+    }
+  }
+
+  function disableNext() {
+    const ind = steps.indexOf(currentStep);
+    return stepsState[ind + 1] === 'disabled';
+  }
+
+  function resetControllerInfo(accountId: string | null) {
+    setControllerAlreadyBonded(false);
+    setControllerAccountId(accountId);
+  }
+
+  /**
+   * double fees for case if fees will be changed on a small count of funds
+   * @todo - compare with account balance and throw error if more
+   */
+  function setAmountToTransfer() {
+    const minAmount = new BN(0);
+    setTransferableAmount(
+      minAmount
+        .iadd(wholeFees)
+        .iadd(wholeFees)
+        .isub(controllerBalance || new BN(0))
+    );
+  }
+
+  function calculateMaxPreFilledBalance() {
+    if (accountBalance && wholeFees && !amountToBond) {
+      // double wholeFees
+      setAmountToBond(accountBalance.isub(wholeFees).isub(wholeFees).isub(existentialDeposit));
+    }
+    return 0;
+  }
+
+  function isBalanceEnough() {
+    return (accountBalance
+      && controllerBalance
+      && existentialDeposit
+      && wholeFees
+      && accountBalance.cmp(existentialDeposit) === 1
+      && controllerBalance.cmp(wholeFees) === 1)
+  }
+
+  const _onUpdateControllerState = useCallback(
+    (controllerAlreadyBonded: boolean): void => {
+      setControllerAlreadyBonded(controllerAlreadyBonded);
+    },
+    [controllerAlreadyBonded]
+  );
+
+  const _onUpdateNominatedState = useCallback(
+    (isNominated) => {
+      setIsNominated(isNominated);
+    },
+    [isNominated]
+  );
+
+  /**
+   * Set validators list.
+   * If filtered validators
+   */
+  useEffect(() => {
+    if (validators && validators.length) {
+      setSelectedValidators(
+        validators.map((validator): string => validator.key).slice(0, 16)
+      );
+    } else {
+      stakingOverview && setSelectedValidators(
+        stakingOverview.validators.map((acc): string => acc.toString()).slice(0, 16)
+      );
+    }
+  }, [validators, stakingOverview]);
+
+  useEffect(() => {
+    if (!wholeFees) {
+      return;
+    }
+    setAmountToTransfer();
+    calculateMaxPreFilledBalance();
+  }, [accountBalance, controllerBalance, wholeFees]);
```

Comment:
```suggestion
  useEffect(() => {
    if (!wholeFees) {
      return;
    }
    
    setAmountToTransfer();
    calculateMaxPreFilledBalance();
  }, [accountBalance, controllerBalance, wholeFees]);
```

Block spacing, linting.